module/mempool/consensus/execution_tree_test.go

```diff
@@ -160,17 +161,49 @@ func (et *ExecutionTreeTestSuite) Test_AddReceipt() {
 	assert.Equal(et.T(), uint(2), et.Forest.Size())
 }
 
-// Test_AddResult verifies that vertices can be added to the Execution Tree
-// without requiring an Execution Rereipt
-func (et *ExecutionTreeTestSuite) Test_AddResult() {
-	// TODO: implement me
-	et.T().Skip()
+// Test_AddResult_Detached verifies that vertices can be added to the Execution Tree without requiring
+// an Execution Receipt. Here, we add a result for a completely detached block. Starting a tree search
+// from this result should not yield any receipts.
+func (et *ExecutionTreeTestSuite) Test_AddResult_Detached() {
+	miscBlock := makeBlockWithHeight(101)
+	miscResult := unittest.ExecutionResultFixture(unittest.WithBlock(miscBlock))
+
+	err := et.Forest.AddResult(miscResult, miscBlock.Header)
+	assert.NoError(et.T(), err)
+	collectedReceipts, err := et.Forest.ReachableReceipts(miscResult.ID(), anyBlock(), anyReceipt())
+	assert.NoError(et.T(), err)
+	et.Assert().Empty(collectedReceipts)
+}
+
+// Test_AddResult_Bridge verifies that vertices can be added to the Execution Tree without requiring
+// an Execution Receipt. Here, we add the result r[C12], which closes the gap between r[C11] and r[C13].
+// Hence, the tree search should reach r[C13].
+func (et *ExecutionTreeTestSuite) Test_AddResult_Bridge() {
```

Comment:
Suggestion: first check that there is no path and after adding bridge verify that there is a path

Reply:
👍 added in commit https://github.com/onflow/flow-go/pull/322/commits/2ed4e73f56bea54f3339053be8660d7123b82039